src/lib/work_partitioner/id_generator.ml

```diff
@@ -0,0 +1,24 @@
+open Core_kernel
+
+(* NOTE: range is both inclusive *)
+type t = { logger : Logger.t; range : int64 * int64; mutable next_id : int64 }
+
+let create ~logger =
+  { logger
+  ; range = (Int64.min_value, Int64.max_value)
```

Comment:
Is there a plan to ever have range to be something different? Maybe it makes sense to define this as module constants instead of field of a structure?

Reply:
Make sense, I'll lift these to constants.